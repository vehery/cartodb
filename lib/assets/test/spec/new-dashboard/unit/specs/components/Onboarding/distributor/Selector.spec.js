import { shallowMount } from '@vue/test-utils';
import Selector from 'new-dashboard/components/Onboarding/distributor/Selector';

const $t = key => key;

describe('Selector.vue', () => {
  it('should render properly', () => {
    const componentProps = {
      title: 'JavaScript Tutorial',
      text: 'This a fake description',
      tags: ['Fake tag', 'More fake tag']
    };

    const selectorComponent = shallowMount(Selector, {
      propsData: componentProps,
      mocks: { $t }
    });

    expect(selectorComponent).toMatchSnapshot();
  });

  it('should set icon modifier if present', () => {
    const componentProps = {
      iconModifier: 'fake'
    };

    const selectorComponent = shallowMount(Selector, {
      propsData: componentProps,
      mocks: { $t }
    });

    const iconNode = selectorComponent.find('.title');
    expect(iconNode.classes()).toContain('selector__title-icon', 'selector__title-icon--fake');
  });
});
