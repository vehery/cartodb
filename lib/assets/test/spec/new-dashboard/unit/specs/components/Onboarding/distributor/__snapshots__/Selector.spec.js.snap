// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector.vue should render properly 1`] = `
<div tabindex="0" class="selector">
  <div>
    <h3 class="selector__title title is-body is-medium u-mb--8">
      JavaScript Tutorial
    </h3>
    <p class="text is-caption">This a fake description</p>
  </div>
  <ul class="selector__tags">
    <li class="selector__tags-item title">Fake tag</li>
    <li class="selector__tags-item title">More fake tag</li>
  </ul>
</div>
`;
